source/developer/running-mattermost.md

```diff
@@ -0,0 +1,50 @@
+Running Mattermost
+-----------------------------
+
+To run Mattermost in development mode locally after setting up your local machine, type: `make run`
+
+To stop Mattermost, type: `make stop`
+
+To restart Mattermost after server code changes, type: `make restart-server`
+
+Client code changes are automatically recompiled on saving.
+
+To clean the database:
+  - `make stop`
+  - `make clean-docker`
+  - `make run`
+  
+To run a test SMTP server: `sudo python -m smtpd -n -c DebuggingServer localhost:25`
+
+### Make Commands ###
+
+Here is a complete listing of all `make` commands:
```

Comment:
I don't think we need to document all of these. Most of them are just there to eliminate some duplicated logic. I think we can remove the lines for setup-run-client-tests, cleanup-run-client-tests, .prebuild, prepare-enterprise, build* (they're all part of `make package`), and setup-mac (that's covered in the dev setup instructions)